### PR TITLE
chore: improve PR labeling, fix, sort, add new team member

### PR DESCRIPTION
**Pull request by balazsorban44**
## What?

This PR has its main goal to improve the experience of opening/reviewing a PR.

It improves the PR template, as well as extends PR auto-labeling.

## Why?

There are 2 parts to the PR template:

- Individual contributors: The current template accommodates for the most common cases individual contributors use to contribute. (I.e. fixing bugs, updating/adding examples, addressing documentation issues). The new template keeps this at the top.
- Next.js team: As maintainers, we should try to explain PRs as if someone with less context would review them. This should also help with backreferencing PRs in issues and follow-up PRs and make it easier to follow changes, understand trade-offs better at the time, and help new team members catch up over time, etc.

## How?

The template is written in a comment instead of Markdown, so when the PR is opened, most of the template is hidden to make it less verbose.

The checkboxes we've had before are not shown, but we already auto-label PRs based on the path of the files that have been touched, so the checkboxes showed redundant information.

[Slack thread](https://vercel.slack.com/archives/C04DUD7EB1B/p1677677802109289), Closes NEXT-741

### CODE REVIEW

File: .github/labeler.json [hunk 1]
```diff
@@ -1,53 +1,65 @@
...
+    "area: documentation": ["docs/**", "errors/**"],
+    "area: examples": ["examples/**"],
+    "area: next/image": [
+      "**/*image*",
+      "**/*image*/**",
+      "packages/next/src/client/use-intersection.tsx",
+      "packages/next/src/server/lib/squoosh/",
+      "packages/next/src/server/serve-static.ts"
     ],
+    "area: Font Optimization": ["**/*font*"],
```
Comment (balazsorban44) [inline .github/labeler.json:13]: We could auto-label more features to be able to filter on this while looking for PRs to review, but some features are spread across multiple files that often change, so it's not always easy to keep track of them.

For this reason, I only added `next/image` and `next/font` for now.

File: .github/labeler.json [hunk 2]
```diff
@@ -1,53 +1,65 @@
...
+      { "type": "user", "pattern": "housseindjirdeh" },
       { "type": "user", "pattern": "janicklas-ralph" },
-      { "type": "user", "pattern": "atcastle" },
+      { "type": "user", "pattern": "kara" },
       { "type": "user", "pattern": "kyliau" },
-      { "type": "user", "pattern": "kara" }
+      { "type": "user", "pattern": "spanicker" }
     ],
     "created-by: Next.js team": [
+      { "type": "user", "pattern": "acdlite" },
```
Comment (balazsorban44) [inline .github/labeler.json:25]: Welcome @acdlite :wave:

File: .github/labeler.json [hunk 3]
```diff
@@ -1,53 +1,65 @@
...
-      "packages/react-dev-overlay/**",
-      "packages/react-refresh-utils/**",
-      "packages/next-codemod/**",
-      "packages/eslint-plugin-next/**",
-      "packages/eslint-config-next/**",
-      "packages/next-env/**",
-      "packages/next-swc/**"
+    "area: documentation": ["docs/**", "errors/**"],
+    "area: examples": ["examples/**"],
+    "area: next/image": [
```
Comment (ijjk) [inline .github/labeler.json:6]: Any additional areas we add here we should handle in the changelog generation https://github.com/vercel/next.js/blob/111ef3951cf13e9eb758ace222265fdd3f883d6e/release.js#L3

**Review by styfle — COMMENTED** [review]
Can you do the sort in a separate PR to make this one easier to review? Its hard to tell what is changing.

**Review by timneutkens — APPROVED** [review]

**Pull request merged**